reference/constraints/Compound.rst

```diff
@@ -0,0 +1,110 @@
+Compound
+========
+
+To the contrary to the other constraints, this constraint cannot be used on its own.
+Instead, it allows you to create your own set of reusable constraints, representing
+rules to use consistently across your application, by extending the constraint.
+
+==========  ===================================================================
+Applies to  :ref:`class <validation-class-target>` or :ref:`property or method <validation-property-target>`
+Options     - `groups`_
+            - `payload`_
+Class       :class:`Symfony\\Component\\Validator\\Constraints\\Compound`
+Validator   :class:`Symfony\\Component\\Validator\\Constraints\\CompoundValidator`
+==========  ===================================================================
+
+Basic Usage
+-----------
+
+Suppose that you have different places where a user password must be validated,
+you can create your own named set or requirements to be reused consistently everywhere:
+
+.. configuration-block::
+
+    .. code-block:: php
+
+        // src/Validator/Constraints/MatchesPasswordRequirements.php
+        namespace App\Validator\Constraints;
+
+        use Symfony\Component\Validator\Compound;
+
+        /**
+         * @Annotation
+         */
+        class MatchesPasswordRequirements extends Compound
+        {
+            protected function getConstraints(array $options): array
+            {
+                return [
+                    new NotBlank(),
+                    new Type('string'),
+                    new Length(['min' => 12]),
+                    new NotCompromisedPassword(),
+                ];
+            }
+        }
+
+You can now use it anywhere you need it:
+
+.. configuration-block::
+
+    .. code-block:: php-annotations
+
+        // src/Entity/User.php
+        namespace App\DTO;
+
+        use App\Validator\Constraints\MatchesPasswordRequirements;
```

Comment:
I would do the following consistently with my comment above:
```php
use App\Validator\Constraints\PasswordRequirementContraint as AssertPasswordRequirements;
```

Reply:
We're using `AcmeAssert` for the whole `App\Validator\Constraints` namespace in https://symfony.com/doc/current/validation/custom_constraint.html#using-the-new-validator, maybe we should use it here too for consistency.

(but I personally prefer no alias at all, as there is no ambiguity to me)